ethereum/spec/src/main/java/tech/pegasys/teku/spec/schemas/AbstractSchemaProvider.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.schemas;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.EnumMap;
+import java.util.Map;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.config.SpecConfig;
+import tech.pegasys.teku.spec.schemas.SchemaTypes.SchemaId;
+
+public abstract class AbstractSchemaProvider<T> implements SchemaProvider<T> {
```

Comment:
NIT: we could move this class into a `[..].schemas.providers` package and makes package protected? This way we end up with all new provider implementations in the same package.